api/controllers/user.go

```diff
@@ -184,3 +184,104 @@ func HandleNewCodeValidation(c *gin.Context) {
 
 	c.IndentedJSON(http.StatusOK, gin.H{"message": "New Code created and sended"})
 }
+
+// controller to create and send a code to reset the password
+func HandleCodeResetPassword(c *gin.Context) {
+	var form interfaces.EmailForm
+	if err := c.BindJSON(&form); err != nil {
+		fmt.Println(err)
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"message": "Bad request"})
+		return
+	}
+	//Check if there is no email
+	if form.Email == "" {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"message": "Email cannot be empty"})
+		return
+	}
+
+	_, err := models.GetUserByEmail(form.Email)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusNotFound, gin.H{"message": "This Email has not been registered"})
+		return
+	}
+	//generate code
+	resetPasswordCode := utils.GetValidationCode()
+
+	//update in database reset password code
+	err = models.UpdateResetPassCode(form.Email, resetPasswordCode)
+	if err != nil {
+		fmt.Println(err)
```

Comment:
Please, remove / comment those debug messages when the tasks is fully completed.